### PR TITLE
PRIME-2681 allow duplicate site ID for HA site

### PR DESCRIPTION
**Pull request by bergomi02**

### CODE REVIEW

File: prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.ts [hunk 1]
```diff
@@ -57,11 +58,36 @@ export class SiteOverviewPageComponent implements OnInit {
...
-      this.busy = this.siteResource.updatePecCode(siteId, pec)
-        .subscribe(() => {
-          this.refresh.next(true);
-          this.site.pec = pec;
-        });
+
+      this.siteResource.pecExistsWithinHa(siteId, pec).subscribe((result) => {
+        if (result) {
+          const data: DialogOptions = {
+            title: 'Site ID is in used',
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.ts:65]: Typo, correct to "is in use"

File: prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.ts [hunk 2]
```diff
@@ -57,11 +58,36 @@ export class SiteOverviewPageComponent implements OnInit {
...
+            message: 'This SiteID is already in use.  Allow multiple sites?',
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.ts:66]: Typo, correct to "Site ID"

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/health-authority-table/health-authority-table.component.ts [hunk 3]
```diff
@@ -123,6 +131,10 @@ export class HealthAuthorityTableComponent implements OnInit, OnChanges {
     }
   }
 
+  public getDuplicatePecText(row: HealthAuthoritySiteAdminList) {
+    return `${row.duplicatePecSiteCount + 1} sites share the same site ID.`;
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/health-authority-table/health-authority-table.component.ts:135]: "... Site ID" (without period) better?

File: prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.ts (hunk 2 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.ts]: fixed

File: prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.ts (hunk 1 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/site-overview-page/site-overview-page.component.ts]: Fixed

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/health-authority-table/health-authority-table.component.ts (hunk 3 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/health-authority-table/health-authority-table.component.ts]: Sure, updated.

File: prime-dotnet-webapi/Services/interfaces/ISiteService.cs [hunk 4]
```diff
@@ -13,6 +13,7 @@ public interface ISiteService
         Task<bool> SiteExistsAsync(int siteId);
         Task<SiteStatusType> GetSiteCurrentStatusAsync(int siteId);
         Task<bool> PecAssignableAsync(int siteId, string pec);
+        Task<bool> PecExistsWithinHA(int siteId, string pec);
```
Comment (neophyte57) [inline prime-dotnet-webapi/Services/interfaces/ISiteService.cs:16]: Convention is to suffix these type of method names' with `Async`

File: prime-dotnet-webapi/Controllers/SitesController.cs [hunk 5]
```diff
@@ -726,6 +726,34 @@ public async Task<ActionResult> PecAssignable(int siteId, string pec)
...
         }
 
+        // GET: api/sites/1/pec/abc/exists-within-ha
+        /// <summary>
+        /// Check if a given PEC exists in other site within the same health authority.
+        /// </summary>
+        /// <param name="siteId"></param>
+        /// <param name="pec"></param>
+        /// <returns></returns>
+        [HttpPost("{siteId}/pec/{pec}/exists-within-ha", Name = nameof(PecExistsWithinHA))]
```
Comment (neophyte57) [inline prime-dotnet-webapi/Controllers/SitesController.cs:736]: Semantically, should be a `HttpGet`?

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-admin-site-list.model.ts [hunk 6]
```diff
@@ -34,7 +34,8 @@ export class HealthAuthoritySiteAdminList extends AbstractBaseHealthAuthoritySit
     public isNew: boolean,
     public readonly authorizedUserName: string,
     public readonly authorizedUserEmail: string,
-    public readonly adjudicatorIdir: string
+    public readonly adjudicatorIdir: string,
+    public duplicatePecSiteCount: number
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-admin-site-list.model.ts:38]: This seems like it needs clarification ... It is the number of OTHER sites that have the same Site ID as this row?

File: prime-dotnet-webapi/Services/interfaces/ISiteService.cs (hunk 4 above)
Comment (bergomi02) [reply prime-dotnet-webapi/Services/interfaces/ISiteService.cs]: Fixed

Comment (bergomi02) [reply prime-dotnet-webapi/Services/interfaces/ISiteService.cs]: Fixed

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-admin-site-list.model.ts (hunk 6 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-admin-site-list.model.ts]: Added comment

**Review by neophyte57 — APPROVED** [review]
Nicely done 👍  (just that comment that needs revising)

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-admin-site-list.model.ts [hunk 7]
```diff
@@ -34,7 +34,8 @@ export class HealthAuthoritySiteAdminList extends AbstractBaseHealthAuthoritySit
     public isNew: boolean,
     public readonly authorizedUserName: string,
     public readonly authorizedUserEmail: string,
-    public readonly adjudicatorIdir: string
+    public readonly adjudicatorIdir: string,
+    public duplicatePecSiteCount: number   // number of site sharing the same site ID
```
[the remark this answers is not in the document]
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/shared/models/health-authority-admin-site-list.model.ts]: Updated the comment

**Pull request merged**